### PR TITLE
Add C queries

### PR DESCRIPTION
**Pull request by vigoux**
Also add a missing `field` capture in locals query.

Fixes partially #52.

### CODE REVIEW

File: queries/c/locals.scm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    (enumerator name: (identifier) @definition.variable)))
+
+;; Type / Struct / Enum
+(field_declaration
+  declarator: (field_identifier) @definition.field)
+(type_definition
+  declarator: (type_identifier) @definition.type)
+
+;; References
+(identifier) @reference
```
Comment (kyazdani42) [inline queries/c/locals.scm:31]: shouldn't we define scopes for this too ?

File: queries/c/locals.scm [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+;; Scope
+(for_statement) @scope
```
Comment (kyazdani42) [inline queries/c/locals.scm:34]: we'll add more scopes another time i guess.
You should also update the README though.

Comment (vigoux) [reply queries/c/locals.scm:34]: Actually, C is quite poor in scopes, even `for_statement` is added for convenience, the only real scope used in C is function based I think.

Comment (kyazdani42) [reply queries/c/locals.scm:34]: well to me every condition blocks, while loops, do for loops, switch statement and so on are scopes. But that's ok we'll leave it for another time as i said :)

Comment (vigoux) [reply queries/c/locals.scm:34]: That's the weird part of C, nothing despite functions and files are scopes ;-)

Comment (kyazdani42) [reply queries/c/locals.scm:34]: well i'm not sure but if you do something like (that you shouldnt do though)
```c
if (something) {
  int a = 0;
...
}
```
the a is stored in a register and removed from the register when you are going out of the `if` scope ?

Comment (vigoux) [reply queries/c/locals.scm:34]: You are right, how dumb am I. I'm adding the scopes right now.

**Comment by vigoux** [comment]
Should be better by now

**Pull request merged**